alerts/handler.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 alertmanager-github-receiver Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////////////
+package alerts
+
+import (
+	"encoding/json"
+	"github.com/google/go-github/github"
+	//"github.com/kr/pretty"
+	"github.com/prometheus/alertmanager/notify"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+type ReceiverClient interface {
+	CloseIssue(issue *github.Issue) (*github.Issue, error)
+	CreateIssue(title, body string) (*github.Issue, error)
+	ListOpenIssues() ([]*github.Issue, error)
+}
+
+type ReceiverHandler struct {
+	Client ReceiverClient
+}
+
+// ServeHTTP receives and processes alertmanager notifications. If the alert
+// is firing and a github issue does not yet exist, one is created. If the
+// alert is resolved and a github issue exists, then it is closed.
```

Comment:
Add `TODO(dev): stop auto-closing issues once we are confident that this isn't too spammy`